torch/distributions/kl.py

```diff
@@ -278,6 +299,21 @@ def _kl_laplace_laplace(p, q):
     return t1 + t2 + t3 - 1
 
 
+@register_kl(MultivariateNormal, MultivariateNormal)
+def _kl_multivariatenormal_multivariatenormal(p, q):
+    # From https://en.wikipedia.org/wiki/Multivariate_normal_distribution#Kullback%E2%80%93Leibler_divergence
+    if p.event_shape != q.event_shape:
+        raise ValueError("KL-divergence between two Multivariate Normals with\
+                          different event shapes cannot be computed")
+
+    term1 = _batch_diag(q.scale_tril).log().sum(-1) - _batch_diag(p.scale_tril).log().sum(-1)
```

Comment:
Shouldn't `term1 = 2 * (_batch_diag(q.scale_tril)...)`, since `log(det(cov)) = 2 * log(det(scale_tril))`?

Reply:
I initially went with using `torch.logdet`. I found the `log_det` implementation here https://github.com/pytorch/pytorch/blob/d93d41b2ef0cc3a986aeb3db56935798beaa3f63/torch/distributions/multivariate_normal.py#L188 ,  which I thought was easier to compute.

Reply:
I mean I like the implementation you have, I just think it needs an additional factor of 2. The `log_det` term you referred to is actually `0.5 * log(det(cov))`

Reply:
Sorry again. I fully understand. I think the reason why the tests didn't pick it up was because of the fact that `precision = 0.1`. I have made the change locally, and testing with `0.01` has passed.

Reply:
no need to be sorry ever :smile: